packages/flutter/lib/src/rendering/proxy_box.dart

```diff
@@ -1299,13 +1299,86 @@ class RenderFractionalTranslation extends RenderProxyBox {
   }
 }
 
+/// The interface used by [CustomPaint] (in the widgets library) and
+/// [RenderCustomPaint] (in the rendering library).
+///
+/// To implement a custom painter, subclass this interface to define your custom
+/// paint delegate.
+///
+/// The [paint] method is called whenever the custom object needs to be
+/// repainted. This is the primary method that must be implemented.
```

Comment:
Suggest simplifying the paragraph and using active voice: e.g.
"In your subclass, you must implement the [paint] and [shouldRepaint] methods. The framework calls the [paint] method it needs to repaint the custom object. The framework calls the [shouldRepaint] method when it receives a new instance of a delegate class and needs to check if the new instance represents different information."

Also, could you clarify what is meant by delegate class here?